src/dto/protocol-gas.dto.ts

```diff
@@ -48,19 +50,24 @@ export class ProtocolGasBaseDTO {
     },
   })
   @IsString()
-  // TODO: All the GasTypeCodes used in the historical recornds but not in the List of the available GasTypeCodes in the DB table.
-  // @IsValidCode(GasTypeCode, {
-  //   message: (args: ValidationArguments) => {
-  //     return CheckCatalogService.formatMessage(
-  //       'You reported the value [value] for [fieldname], which is not in the list of valid values for [key].',
-  //       {
-  //         value: args.value,
-  //         fieldname: args.property,
-  //         key: KEY,
-  //       },
-  //     );
-  //   },
-  // })
+  @IsValidCodes(
+    GasComponentCode,
+    (args: ValidationArguments): FindOneOptions<GasComponentCode> => {
+      return { where: { gasComponentCode: In(args.value), groupCode: 'GTC' } };
```

Comment:
You do not need to add a where clause. API needs to allow all the codes.

Reply:
I guess it needs to an array of sting `codes` to validate using this pipe. But you are passing a comma delimited list.